core/src/edu/rice/rbox/Replication/ReplicaManagerGrpc.java

```diff
@@ -151,7 +174,20 @@ public void handleSubscribe(Rbox.SubscribeRequest request,
 
         @Override
         public void handleUnsubscribe(Rbox.UnsubscribeRequest request, StreamObserver<Empty> responseObserver) {
-            // TODO: Remove from subscribers
+            logger.info("Handling unsubscribe request...");
+
+            GameObjectUUID replicaObjectUUID = getGameObjectUUIDFromMessage(request.getMsg());
+            GameObjectUUID primaryObjectUUID = publishers.get(replicaObjectUUID).getGameObjectUUID();
```

Comment:
The `replicaObjectUUID` in your `UnsubscribeRequest` is the uuid of the replica object on the sender superpeer, while `publishers` is mapping replicas on this receiver superpeer to their primaries. So you cannot get the corresponding primary object uuid from `publishers.get(replicaObjectUUID)`.

Reply:
Wait, I was assuming replica and primary objects have different `GameObjectUUID`. Is that correct?